tree/tree/src/TBranch.cxx

```diff
@@ -2592,51 +2596,59 @@ Int_t TBranch::WriteBasket(TBasket* basket, Int_t where)
       fEntryOffsetLen = 2*nevbuf; // assume some fluctuations.
    }
 
-   Int_t nout  = basket->WriteBuffer();    //  Write buffer
-   fBasketBytes[where]  = basket->GetNbytes();
-   fBasketSeek[where]   = basket->GetSeekKey();
-   Int_t addbytes = basket->GetObjlen() + basket->GetKeylen();
-   TBasket *reusebasket = 0;
-   if (nout>0) {
-      // The Basket was written so we can now safely reuse it.
-      fBaskets[where] = 0;
+   auto do_updates = [=]() {
+      Int_t nout  = basket->WriteBuffer();    //  Write buffer
+      if (nout < 0) {Error("TBranch::WriteBasketImpl", "basket's WriteBuffer failed.\n");}
+      fBasketBytes[where]  = basket->GetNbytes();
+      fBasketSeek[where]   = basket->GetSeekKey();
+      Int_t addbytes = basket->GetObjlen() + basket->GetKeylen();
+      TBasket *reusebasket = 0;
+      if (nout>0) {
+         // The Basket was written so we can now safely reuse it.
+         fBaskets[where] = 0;
 
-      reusebasket = basket;
-      reusebasket->Reset();
+         reusebasket = basket;
+         reusebasket->Reset();
```

Comment:
No tab characters here; that's just how github renders the added spaces, unfortunately.